horovod/common/util.py

```diff
@@ -265,3 +265,24 @@ def is_iterable(x):
     except TypeError:
         return False
     return True
+
+
+@_cache
+def is_version_greater_equal_than(ver, target):
```

Comment:
Do we actually need a function for version comparison? It looks like the whole thing is duplicating what StrictVersion is doing.

Reply:
this is from @DEKHTIARJonathan :) I cherry-picked his work.

Reply:
yea i think just comparing using StrictVersion is sufficient in your case.

Reply:
After revisiting the code, I think this function to cache `TF` version results instead of introducing a new function on top of `LooseVersion`.

Since there could be multiple places in a program running life cycle to check TF version >= 2.6.

Reply:
i see so it's saving results for different version comparisons. then let's keep it, remove the StrictVersion import though, it's not being used anywhere.